app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -527,70 +529,76 @@ class HomeFragment : Fragment(), CoroutineScope, AccountObserver {
-                    .filter { (activity as HomeActivity).browsingModeManager.isPrivate == it.private }
-                    .filter { it.id != sessionId }
-                    .map {
-                        val selected =
-                            it == sessionManager.selectedSession
-                        Tab(
-                            it.id,
-                            it.url,
-                            it.url.urlToTrimmedHost(),
-                            it.title,
-                            selected,
-                            it.thumbnail
-                        )
-                    }
+        launch(Dispatchers.Main) {
+            val sessionManager = requireComponents.core.sessionManager
+
+            // Update the UI with the tab removed, but don't remove it from storage yet
+            getManagedEmitter<SessionControlChange>().onNext(
+
+                SessionControlChange.TabsChange(
+                    sessionManager.sessions
+                        .filter { (activity as HomeActivity).browsingModeManager.isPrivate == it.private }
+                        .filter { it.id != sessionId }
+                        .map {
+                                val selected = it == sessionManager.selectedSession
+                                Tab(
+                                    it.id,
+                                    it.url,
+                                    it.url.urlToTrimmed(context!!),
+                                    it.title,
+                                    selected,
+                                    it.thumbnail
+                            )
+                        }
+                )
             )
-        )
-
-        deleteSessionJob = {
-            sessionManager.findSessionById(sessionId)
-                ?.let { session ->
-                    sessionManager.remove(session)
-                }
-        }
 
-        allowUndo(
-            view!!, getString(R.string.snackbar_tab_deleted),
-            getString(R.string.snackbar_deleted_undo), {
-                deleteSessionJob = null
-                emitSessionChanges()
+            deleteSessionJob = {
+                sessionManager.findSessionById(sessionId)
+                    ?.let { session ->
+                        sessionManager.remove(session)
+                    }
             }
-        ) {
-            sessionManager.findSessionById(sessionId)
-                ?.let { session ->
-                    sessionManager.remove(session)
+
+            allowUndo(
+                view!!, getString(R.string.snackbar_tab_deleted),
+                getString(R.string.snackbar_deleted_undo), {
+                    deleteSessionJob = null
+                    emitSessionChanges()
                 }
+            ) {
+                sessionManager.findSessionById(sessionId)
+                    ?.let { session ->
+                        sessionManager.remove(session)
+                    }
```

Comment:
Nit: put these 4 lines of code into a closure `val` and use it here and `deleteSessionJob` above.

Reply:
👍